torch/lib/TH/generic/THBlas.c

```diff
@@ -18,7 +18,18 @@ TH_EXTERNC void scopy_(int *n, float *x, int *incx, float *y, int *incy);
 TH_EXTERNC void daxpy_(int *n, double *a, double *x, int *incx, double *y, int *incy);
 TH_EXTERNC void saxpy_(int *n, float *a, float *x, int *incx, float *y, int *incy);
 TH_EXTERNC double ddot_(int *n, double *x, int *incx, double *y, int *incy);
+#ifdef __APPLE__
+TH_EXTERNC ffloat cblas_sdot(const int n, const float *x, const int incx, const float *y, const int incy);
+#ifndef THBlas_ACCELERATE_sdot_
```

Comment:
Where is this defined? `THBlas_ACCELERATE_sdot_`? is this even defined? I couldn't find it in the codebase.

i think this can be cleaned up properly.
First, TH/CMakeLists.txt:308, add:
```
ELSEIF(BLAS_INFO STREQUAL "accelerate")
    ADD_DEFINITIONS(-DTH_BLAS_ACCELERATE)
```

Then, this section can be cleaned up with:


```
#ifdef TH_BLAS_ACCELERATE
TH_EXTERNC ffloat cblas_sdot(const int n, const float *x, const int incx, const float *y, const int incy);
inline ffloat sdot_(const int *n, const float *x, const int *incx, const float *y, const int *incy)
{
    return cblas_sdot(*n, x, *incx, y, *incy);
}
#else
TH_EXTERNC ffloat sdot_(int *n, float *x, int *incx, float *y, int *incy);
#endif
```